components/ManagerSubmitButton.tsx

```diff
@@ -0,0 +1,54 @@
+import urlsForRouter from "helpers/urlsForRouter";
+import { Button } from "lbh-frontend-react";
+import { NextRouter, useRouter } from "next/router";
+import PropTypes from "prop-types";
+import React from "react";
+import PageSlugs, { urlObjectForSlug } from "steps/PageSlugs";
+
+interface Props {
+  onSubmit(router: NextRouter): Promise<void>;
+  buttonText: string;
+}
+
+const ManagerSubmitButton: React.FunctionComponent<Props> = (props) => {
+  const { onSubmit, buttonText } = props;
+  const router = useRouter();
+  const status = buttonText === "Approve" ? "approved" : "declined";
+
+  const { href, as } = urlsForRouter(router, {
+    ...urlObjectForSlug(router, PageSlugs.Confirmed),
+    query: { status: status },
+  });
+
+  return (
+    <Button
+      preventDoubleClick
+      className={
+        buttonText === "Decline"
+          ? "lbh-button--warning govuk-button--warning"
+          : "submit-button"
+      }
+      onClick={async (): Promise<void> => {
+        if (!href.pathname || !as.pathname) {
+          return;
+        }
+
+        try {
+          await onSubmit(router);
+          await router.push(href, as);
```

Comment:
Let's pull the router push outside the `try..catch` to be consistent with the behaviour of `makeSubmit`. Might even be worth refactoring that's `onClick` method to be used here. I'd expect them to look pretty similar.

Reply:
They do look similar, do you mean extract the `onClick` into more of a shared function, or use `makeSubmit` itself? Obviously the former is a bit less work, although I'm unsure where it would go - a helper maybe?

Reply:
I mean pull out the common logic into something more easily shared.